src/AzSK.ADO/Framework/Core/SVT/SVTResourceResolver.ps1

```diff
@@ -157,6 +162,77 @@ class SVTResourceResolver: AzSKRoot {
+            if($projectData['build'] -eq -1) {
+                $resourceURL = ""
+                if ([string]::IsNullOrEmpty($topNQueryString)) {
+                    $topNQueryString = '&$top=10000'
+                    $resourceURL = ("https://dev.azure.com/{0}/{1}/_apis/build/definitions?api-version=4.1&queryOrder=lastModifiedDescending" +$topNQueryString) -f $($this.SubscriptionContext.SubscriptionName), $projectName;
+                }
+                else {
+                    $resourceURL = ("https://dev.azure.com/{0}/{1}/_apis/build/definitions?api-version=4.1" +$topNQueryString) -f $($this.SubscriptionContext.SubscriptionName), $projectName;                               
+                }
+                $responseList = [WebRequestHelper]::InvokeGetWebRequest($resourceURL);
+                $projectData['build'] = $responseList.Length;
+            }
+            # fetch the release count if provided builds not equals to *
+            if($projectData['release'] -eq -1) {
+                $resourceURL = ("https://vsrm.dev.azure.com/{0}/{1}/_apis/release/definitions?api-version=4.1-preview.3" +$topNQueryString) -f $($this.SubscriptionContext.SubscriptionName), $projectName;
+                $responseList = [WebRequestHelper]::InvokeGetWebRequest($resourceURL);
+                $projectData['release'] = $responseList.Length;
+            }
+            # fetch the agent pools count
+            if($projectData["agentpools"] -eq -1) {
+                $agentPoolsDefnURL = ("https://dev.azure.com/{0}/{1}/_settings/agentqueues?__rt=fps&__ver=2") -f $($this.SubscriptionContext.SubscriptionName), $projectName;
+                try {
+                    $agentPoolsDefnsObj = [WebRequestHelper]::InvokeGetWebRequest($agentPoolsDefnURL);
+                    if (([Helpers]::CheckMember($agentPoolsDefnsObj, "fps.dataProviders.data") ) -and (($agentPoolsDefnsObj.fps.dataProviders.data."ms.vss-build-web.agent-queues-data-provider") -and $agentPoolsDefnsObj.fps.dataProviders.data."ms.vss-build-web.agent-queues-data-provider".taskAgentQueues)) {
+                        $nObj = $this.MaxObjectsToScan
+                        $taskAgentQueues = $null;
+                        $taskAgentQueues = $agentPoolsDefnsObj.fps.dataProviders.data."ms.vss-build-web.agent-queues-data-provider".taskAgentQueues | where-object{$_.pool.isLegacy -eq $false};
+                        if ($null -ne $taskAgentQueues) {
+                            $projectData["agentpools"] = $taskAgentQueues.Length
+                        }
+                    }
+                }
+                catch {}
+            }
+        }
+        catch {}
+        # fetch the variable groups count
+        if ($projectData["variablegroups"] -eq -1) {
+            $variableGroupURL = ("https://dev.azure.com/{0}/{1}/_apis/distributedtask/variablegroups?api-version=6.1-preview.2") -f $($this.organizationName), $projectId;
+            try{
```

Comment:
why is this added in a separate try-catch?

it is better then to segregate each resource type computation in individual try-catch blocks.